PhysicsTools/NanoAOD/plugins/LeptonTimeLifeInfoTableProducer.cc

```diff
@@ -0,0 +1,340 @@
+/**
+  \class    LeptonTimeLifeInfoTableProducer
+  \brief    Produces FlatTable with lepton life-time information
+
+  \author   Michal Bluj, NCBJ, Warsaw
+*/
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/NanoAOD/interface/FlatTable.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "RecoVertex/VertexPrimitives/interface/ConvertToFromReco.h"
+#include "RecoVertex/KalmanVertexFit/interface/KalmanVertexFitter.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/GeomPropagators/interface/AnalyticalTrajectoryExtrapolatorToLine.h"
+#include "TrackingTools/GeomPropagators/interface/AnalyticalImpactPointExtrapolator.h"
+#include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
+#include "Utilities/General/interface/ClassName.h"
+#include "PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h"
+#include "DataFormats/NanoAOD/interface/LeptonTimeLifeInfo.h"
+
+#include <cstring>
+
+namespace {
```

Comment:
so as to avoid redefining this outside of the templated table producer, please split the producer in two. One that creates `LeptonTimeLifeInfo`, one that reads it back from edm to fill a table

Reply:
It is in fact thins is what I wanted to avoid. The idea is to create a nanoAOD table directly without using intermediate products put to and then read from `edm::Event`. 
The `LeptonTimeLifeInfo` is meant to be an internal data structure rather than (potentially) persistent data-format, but to be able to use string parser to configure variables to be stored in the table the structure of the `LeptonTimeLifeInfo`  has to be parsed by reflex. So it is why is is defined outsite in DataFormats and added to classes.h/_def.xml. 
BTW, in the first version of this development intermediate collections of pat::Electron/Muon/Tau with set of `userFloat`s were produced, but I found it (i) not very flexible due to hard-coded names of the floats, (ii) not optimal due to need of producing quite big objects containing mainly duplicated information from original leptons. I can revert to it if it is preferred.